### PR TITLE
Add new requirements for stub in module guidelines

### PR DESCRIPTION
**Pull request by jfy133**
As discussed in office hours

### CODE REVIEW

File: sites/docs/src/content/docs/guidelines/components/modules.md [hunk 1]
```diff
@@ -324,6 +324,34 @@ See the [Bash manual on file operators](https://tldp.org/LDP/abs/html/fto.html)
...
 
+### Stubs
+
+#### Stub block must exist
+
+A stub block MUST exist for all modules
+
+#### Stub files for all output channels 
+
+The stub block MUST include the creation of a file for every output channel (both mandatory and optional), generated with touch, e.g.
```
Comment (jfy133) [inline sites/docs/src/content/docs/guidelines/components/modules.md:335]:
```suggestion
The stub block MUST include the creation of at least one file for every output channel (both mandatory and optional), generated with touch, e.g.
```

**Review by nvnieuwk — APPROVED** [review]
LGTM!

File: sites/docs/src/content/docs/guidelines/components/modules.md [hunk 2]
```diff
@@ -324,6 +324,34 @@ See the [Bash manual on file operators](https://tldp.org/LDP/abs/html/fto.html)
...
+
+A stub block MUST exist for all modules
+
+#### Stub files for all output channels 
+
+The stub block MUST include the creation of at least one file for every output channel (both mandatory and optional), generated with touch, e.g.
+
+```bash
+touch ${prefix}.txt
+```
```
Comment (maxulysse) [inline sites/docs/src/content/docs/guidelines/components/modules.md:339]: I would add that the sub block should reproduce as much as possible the number and filenames that are expected as output

**Review by mirpedrol — APPROVED** [review]
> We should also open an issue to nf-core/tools to add a linting test for this.

which bit? i can make an issue

File: sites/docs/src/content/docs/guidelines/components/modules.md [hunk 3]
```diff
@@ -324,6 +324,34 @@ See the [Bash manual on file operators](https://tldp.org/LDP/abs/html/fto.html)
...
+
```
Comment (jfy133) [inline sites/docs/src/content/docs/guidelines/components/modules.md:340]:
```suggestion

Ideally, the stub block should reproduce as much as possible the number of, and filenames, of the files expected as output.

```

File: sites/docs/src/content/docs/guidelines/components/modules.md (hunk 2 above)
Comment (jfy133) [reply sites/docs/src/content/docs/guidelines/components/modules.md:339]: Done

File: sites/docs/src/content/docs/guidelines/components/modules.md [hunk 4]
```diff
@@ -324,6 +324,36 @@ See the [Bash manual on file operators](https://tldp.org/LDP/abs/html/fto.html)
 
 Alternate suggestions include using `grep -c` to search for a valid string match, or other tool which will appropriately error when the expected output is not successfully created.
 
+### Stubs
+
+#### Stub block must exist
+
+A stub block MUST exist for all modules
```
Comment (SPPearce) [inline sites/docs/src/content/docs/guidelines/components/modules.md:331]:
```suggestion
[A stub block](https://www.nextflow.io/docs/latest/process.html#stub) MUST exist for all modules. This is a block of code that replaces the `script` command when the option `-stub` is set. This enables quick testing of the workflow logic, as a "dry-run".
```

File: sites/docs/src/content/docs/guidelines/components/modules.md [hunk 5]
```diff
@@ -324,6 +324,36 @@ See the [Bash manual on file operators](https://tldp.org/LDP/abs/html/fto.html)
...
+
+#### Stub files for all output channels 
+
+The stub block MUST include the creation of at least one file for every output channel (both mandatory and optional), generated with touch, e.g.
+
+```bash
+touch ${prefix}.txt
+```
```
Comment (SPPearce) [inline sites/docs/src/content/docs/guidelines/components/modules.md:339]:
```suggestion
```bash
stub:
"""
touch ${prefix}.txt
"""
```
```

File: sites/docs/src/content/docs/guidelines/components/modules.md [hunk 6]
```diff
@@ -324,6 +324,36 @@ See the [Bash manual on file operators](https://tldp.org/LDP/abs/html/fto.html)
...
+
+#### Stub gzip files must use echo and pipe
+
+Stub files that should be output as gzip compressed, MUST use the syntax in the following example:
+
+```bash
+echo "" | gzip > ${prefix}.txt.gz
+```
+
+:::info{title="Rational" collapse}
```
Comment (mashehu) [inline sites/docs/src/content/docs/guidelines/components/modules.md:351]:
```suggestion
:::info{title="Rationale" collapse}
```

File: sites/docs/src/content/docs/guidelines/components/modules.md [hunk 7]
```diff
@@ -324,6 +324,38 @@ See the [Bash manual on file operators](https://tldp.org/LDP/abs/html/fto.html)
 
 Alternate suggestions include using `grep -c` to search for a valid string match, or other tool which will appropriately error when the expected output is not successfully created.
 
+### Stubs
+
+#### Stub block must exist
+
+[A stub block](https://www.nextflow.io/docs/latest/process.html#stub) MUST exist for all modules. This is a block of code that replaces the `script` command when the option `-stub` is set. This enables quick testing of the workflow logic, as a "dry-run".
+
+#### Stub files for all output channels
```
Comment (jfy133) [inline sites/docs/src/content/docs/guidelines/components/modules.md:333]:
```suggestion

#### Stub block prefix and versions

The stub block MUST include the same variables (e.g. `prefix`) and HEREDOC code as the main script block. 

#### Stub files for all output channels 
```

**Review by mashehu — APPROVED** [review]

File: sites/docs/src/content/docs/guidelines/components/modules.md [hunk 8]
```diff
@@ -328,15 +328,17 @@ Alternate suggestions include using `grep -c` to search for a valid string match
...
 
 The stub block MUST include the creation of at least one file for every output channel (both mandatory and optional), generated with touch, e.g.
 
-```bash
+```groovy
+stub:
+"""
 touch ${prefix}.txt
-```
+"""
```
Comment (mashehu) [inline sites/docs/src/content/docs/guidelines/components/modules.md:345]:
````suggestion
"""
```
````

**Review by maxulysse — APPROVED** [review]

**Comment by mashehu** [comment]
@nf-core-bot fix linting

**Pull request merged**